processor/elasticprocessor/internal/hostmetrics/process.go

```diff
@@ -0,0 +1,105 @@
+package hostmetrics
+
+import (
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+)
+
+func addProcessMetrics(metrics pmetric.MetricSlice, dataset string) error {
+	var timestamp pcommon.Timestamp
+	var threads, memUsage, memVirtual, fdOpen int64
+	var memUtil, total, cpuTimeValue float64
+
+	for i := 0; i < metrics.Len(); i++ {
+		metric := metrics.At(i)
+		if metric.Name() == "process.threads" {
+			dp := metric.Sum().DataPoints().At(0)
+			timestamp = dp.Timestamp()
+			threads = dp.IntValue()
+		} else if metric.Name() == "process.memory.utilization" {
+			dp := metric.Gauge().DataPoints().At(0)
+			timestamp = dp.Timestamp()
```

Comment:
we are overriding timestamp for each case here. we can just set in the first case and assume all the datapoints have the same timestamp.

Reply:
there are lots of assumptions in the processor about the format of the input data. we know that these metrics are contained in the same `ScopeMetrics`. As long as we pull the timestamp from a default metric, not one that has to be enabled in config, i think we are fine.

Reply:
@tommyers-elastic : Actually,  there is a possibility that a user can disable a particular default metric as well by doing this. 

```
metrics:
  <metric_name>:
    enabled: false
```

In that case, if the metric where we specify timestamp is disabled, we will not get timestamp for other metrics as well, I suppose